### PR TITLE
Implement changes to implement cursors in go-ase

### PR DESCRIPTION
**Pull request by ntnn**
**Description**

Implement changes to implement cursors in go-ase.

**Related issues**

Link any related issues here.

**Tests**

- [x] make lint
- [x] make test

### CODE REVIEW

File: tds/packageCurOpen.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	if err := ch.WriteByte(byte(TDS_CUROPEN)); err != nil {
+		return err
+	}
+
+	// 4 cursorid
+	// if cursorId == 0:
+	//   1 name len
+	//   x name
+	// 1 status
+	totalLength := 5
```
Comment (fwilhelm92) [inline tds/packageCurOpen.go:83]: for the sake of uniformity this could be written as
`totalLength := 4 + 1`

File: tds/packageKey.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package tds
+
+import (
+	"fmt"
+
+	"github.com/SAP/go-dblib/asetypes"
+)
+
```
Comment (fwilhelm92) [inline tds/packageKey.go:13]: Missing type-description/doc?

File: tds/packageKey.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+type KeyPackage struct {
+	DataType asetypes.DataType
+	Value    interface{}
+}
+
```
Comment (fwilhelm92) [inline tds/packageKey.go:18]: `// ReadFrom implements the tds.Package interface.`

File: tds/packageKey.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	pkg.Value, err = pkg.DataType.GoValue(endian, bs)
+	if err != nil {
+		return fmt.Errorf("tds: error converting bytes to %s: %w", pkg.DataType, err)
+	}
+
+	return nil
+}
+
```
Comment (fwilhelm92) [inline tds/packageKey.go:46]: `// WriteTo implements the tds.Package interface.`

File: tds/packageKey.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+		}
+	}
+
+	if err := ch.WriteBytes(bs); err != nil {
+		return err
+	}
+
+	return nil
+}
+
```
Comment (fwilhelm92) [inline tds/packageKey.go:70]: `// String implements the fmt.Stringer interface.`

**Review by fwilhelm92 — APPROVED** [review]

**Review by fwilhelm92 — APPROVED** [review]

**Review by cb80 — APPROVED** [review]

**Pull request merged**